### PR TITLE
MWPW-163041: analytics fix

### PR DESCRIPTION
**Pull request by 3ch023**
https://jira.corp.adobe.com/browse/MWPW-163041

In this PR:

- add documentation for analytics attributes: see https://mwpw-163041--milo--adobecom.aem.page/libs/features/mas/docs/merch-card.html and https://mwpw-163041--milo--adobecom.aem.page/libs/features/mas/docs/checkout-link.html
- fix daa-ll regression
- add additional unit-test coverage

**Test URLs:**
- Before: https://main--milo--adobecom.aem.page/?martech=off
- After: https://MWPW-163041--milo--adobecom.aem.page/libs/features/mas/docs/mas.js.html?martech=off

### CODE REVIEW

File: libs/features/mas/commerce/checkout-link.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ See [MAS](mas.html#terminology) to learn more.
...
 | `data-modal`                  | `modal` flag for client side interpretation                                                                                                                                                                                                  | `false`       | `false`  |
+| `data-analytics-id`           | human-readable, non-translatable link id for analytics. Authored in Studio in Link Editor.                                                                                                                                                                                                  | `false`       | `false`  |
+| `daa-ll`                      | martech-compatible link id for analytics. Format: '${data-analytics-id}-${#}', where # is the position of the link within a card. E.g. : see-terms-1,  buy-now-2                                                                                                                                                                  | `false`       | `false`  |
```
Comment (npeltier) [inline libs/features/mas/commerce/checkout-link.md:42]: sorry to jump on this on your PR but we should probably somehow make a differentiation in documenting attributes that are inputs and those that are outputs. or "generated"? 
looks like spectrum is not making the difference, see `pending` in https://opensource.adobe.com/spectrum-web-components/components/picker/api/ but that does not mean it is right :)

File: libs/features/mas/docs/merch-card.html [hunk 2]
```diff
@@ -145,6 +145,12 @@ <h3 id="attributes" tabindex="-1">Attributes <a class="header-anchor" href="#att
...
 </tr>
+<tr>
+<td><code>daa-lh</code></td>
+<td>Analytics identifier of a card. Value is coming from the ‘PRODUCT_CODE’ tag set on the card (for now manual authoring required, later will be tagged automatically). Sample values: ‘ccsn’ or ‘phlt’.</td>
+<td></td>
+<td><code>false</code></td>
+</tr>
 </tbody>
 </table>
 <h4 id="active-variants%3A" tabindex="-1">Active variants: <a class="header-anchor" href="#active-variants%3A" href="#active-variants%3A" title="Permalink to this heading">#</a></h4>
```
Comment (npeltier) [inline libs/features/mas/docs/merch-card.html:156]: not sure where that double href is coming from (again, not your PR)

**Review by npeltier — APPROVED** [review]
lgtm with minor comments

File: libs/features/mas/docs/merch-card.html (hunk 2 above)
Comment (3ch023) [reply libs/features/mas/docs/merch-card.html]: double hrefs?

File: nala/features/masccd/masccd.spec.js [hunk 3]
```diff
@@ -37,6 +37,11 @@ module.exports = {
         linkText: 'See terms',
         linkUrl: '',
         iconUrl: '',
+        cardDaaLH: 'ccsn',
+        linkAnalyticsId: 'see-terms',
+        linkDaaLL: 'see-terms-2',
+        ctaAnalyticsId: 'buy-now',
+        ctaDaaLL: 'buy-now-2',
```
Comment (afmicka) [inline nala/features/masccd/masccd.spec.js:44]: @3ch023 this value here should be "buy-now-1". The tests fails

Comment (3ch023) [reply nala/features/masccd/masccd.spec.js]: thx,done

**Pull request merged**